docs/en/ingest-management/processors/processor-decode_base64_field.asciidoc

```diff
@@ -0,0 +1,56 @@
+[[decode_base64_field-processor]]
+= Decode Base64 fields
+
+++++
+<titleabbrev>decode_base64_field</titleabbrev>
+++++
+
+The `decode_base64_field` processor specifies a field to base64 decode.
+
+To overwrite fields, either rename the target field or use the `drop_fields`
+processor to drop the field, and then rename the field.
+
+[discrete]
+== Example
+
+In this example, `field1` is decoded in `field2`.
+
+[source,yaml]
+----
+  - decode_base64_field:
+      field:
+        from: "field1"
+        to: "field2"
+      ignore_missing: false
+      fail_on_error: true
+----
+
+[discrete]
+== Configuration settings
+
+[options="header"]
+|===
+| Name | Required | Default | Description
+
+| `field`
+| Yes
+|
+a| Contains:
+
+* `from: "old-key"`, where `from` is the origin
+* `to: "new-key"`, where `to` is the target field name
+
+| `ignore_missing`
+| No
+| `false`
+| Whether to ignore missing keys. If `true`, missing keys that should be base64 decoded are ignored and no error is logged.
+
+| `fail_on_error`
+| No
+| `true`
+| Whether to fail if an error occurs. If `true` and an error occurs, base64 decoding of fields is stopped, and the original event is returned. If `false`, decoding continues even if an error occurs during decoding.
+|===
+
+See <<conditions>> for a list of supported conditions.
+
+//REVIEWERS: What happens if ignore_missing is false and there are missing keys?
```

Comment:
Need input from reviewers on this question.

Reply:
In this case, the behavior of `fail_on_error` applied. From my reading of the code, if `fail_on_error` is `true` an error will be logged and the event will be dropped, if `fail_on_error` is `false` then an error is logged but the event is not modified.

I think the description of `fail_on_error` may need to be adjusted?